### ISSUE TITLE
When working with assumed role, the underlying execution role is being used for the plugin

### ISSUE THREAD

**Issue by enr0c** [aws/boto3-s3-access-grants-plugin#13]
## Situation:
Working in Sagemaker Studio, Jupyterlab, Python Kernel
The issue seems to be that the plugin does not take into consideration the assumed role, but the execution role of sagemaker. 

## Issue:

```
import boto3
from aws_s3_access_grants_boto3_plugin.s3_access_grants_plugin import S3AccessGrantsPlugin

region = 'region'
s3bucket = 's3bucket'

session = boto3.Session()

sts = session.client("sts")
print("before: ",boto3.client('sts').get_caller_identity()['Arn'])
response = sts.assume_role(
    RoleArn="arn:aws:iam::0123456789012:role/AssumedRole,
    RoleSessionName="AssumedRole"
)

new_session = boto3.Session(aws_access_key_id=response['Credentials']['AccessKeyId'],
                      aws_secret_access_key=response['Credentials']['SecretAccessKey'],
                      aws_session_token=response['Credentials']['SessionToken'])

sts_client = new_session.client('sts')
print("after: ",sts_client.get_caller_identity()['Arn'])


s3_client_boto3 = new_session.client('s3',region_name=region)
plugin = S3AccessGrantsPlugin(s3_client=s3_client_boto3, fallback_enabled=False)
plugin.register()

try:
    s3_client_boto3.get_object(Bucket=s3bucket, Key='lorem/ipsum.json')

except Exception as e: 
    print(e)
```

Output will be:
```
- before:  arn:aws:sts::0123456789012:assumed-role/BeforeRole
- after:  arn:aws:sts::0123456789012:assumed-role/AssumedRole
- An error occurred (AccessDenied) when calling the GetDataAccess operation: You do not have READ permissions to the requested S3 Prefix: s3://s3bucket/lorem/ipsum.json

```

S3 Access grantes were only granted to the assumed role 'arn:aws:sts::0123456789012:assumed-role/AssumedRole'

## Expected behavior
Take the assumed role as input and determine based on the assumed role, if the role has access via S3 access grants to the object/path or not

**Comment by nikitamsawant** [comment]
Hi enr0c,
Thank you for bringing this to our attention.  We were able to reproduce the issue, and it was a bug.  This should be patched as of release 1.1.0.  Can you please upgrade and confirm your problem is resolved?

**Comment by enr0c** [comment]
works like charm now. Feel free to close the ticket and thank you very much again!
